hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/NodeQueryCacheEventService.java

```diff
@@ -119,17 +115,15 @@ public String addListener(String mapName, String cacheId, MapListener listener,
 
     @Override
     public boolean removeListener(String mapName, String cacheId, String listenerId) {
-        String listenerName = generateListenerName(mapName, cacheId);
-        return eventService.deregisterListener(SERVICE_NAME, listenerName, listenerId);
+        return eventService.deregisterListener(SERVICE_NAME, cacheId, listenerId);
```

Comment:
Unrelated to this PR: is it OK that `removeAllListeners` is guarded with a mutex from `lifecycleMutexFactory` but `removeListener` is not?